app/src/test/java/br/com/catbag/gifreduxsample/reducers/middlewares/RestMiddlewareTest.java

```diff
@@ -0,0 +1,213 @@
+package br.com.catbag.gifreduxsample.reducers.middlewares;
+
+import com.umaplay.fluxxan.Action;
+import com.umaplay.fluxxan.Fluxxan;
+import com.umaplay.fluxxan.Middleware;
+import com.umaplay.fluxxan.impl.BaseMiddleware;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+import org.robolectric.util.Logger;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+import br.com.catbag.gifreduxsample.BuildConfig;
+import br.com.catbag.gifreduxsample.actions.PayloadParams;
+import br.com.catbag.gifreduxsample.asyncs.data.DataManager;
+import br.com.catbag.gifreduxsample.asyncs.net.downloader.FileDownloader;
+import br.com.catbag.gifreduxsample.middlewares.RestMiddleware;
+import br.com.catbag.gifreduxsample.models.AppState;
+import br.com.catbag.gifreduxsample.models.Gif;
+import br.com.catbag.gifreduxsample.models.ImmutableAppState;
+import shared.ReduxBaseTest;
+import shared.TestHelper;
+
+import static br.com.catbag.gifreduxsample.actions.GifActionCreator.GIF_DOWNLOAD_FAILURE;
+import static br.com.catbag.gifreduxsample.actions.GifActionCreator.GIF_DOWNLOAD_START;
+import static br.com.catbag.gifreduxsample.actions.GifActionCreator.GIF_DOWNLOAD_SUCCESS;
+import static br.com.catbag.gifreduxsample.actions.GifListActionCreator.GIF_LIST_FETCHING;
+import static br.com.catbag.gifreduxsample.actions.GifListActionCreator.GIF_LIST_UPDATED;
+import static br.com.catbag.gifreduxsample.asyncs.data.DataManager.GifListLoadListener;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static shared.TestHelper.buildGif;
+
+// Roboeletric still not supports API 24 stuffs
+@Config(sdk = 23, constants=BuildConfig.class)
+@RunWith(RobolectricTestRunner.class)
+public class RestMiddlewareTest extends ReduxBaseTest {
+
+    public static final boolean EXPECTED_HAS_MORE = false;
+    public static final List EXPECTED_LIST = new ArrayList<>();
+
+    private Fluxxan<AppState> mFluxxan;
+    private CountDownLatch mSignal;
+    private Action mLastAction;
+
+    private Middleware mInterceptMiddleware = new BaseMiddleware() {
+        @Override
+        public void intercept(Object o, Action action) throws Exception {
+            mLastAction = action;
+            mSignal.countDown();
+        }
+    };
+
+    public RestMiddlewareTest() {
+        AppState state = ImmutableAppState.builder().build();
```

Comment:
Put the ImmutableAppState.builder().build(); directly on the constructor if you want...